src/components/EnhancedMap/ImageMarkerLayer/ImageMarkerLayer.jsx

```diff
@@ -126,29 +91,43 @@ const MapillaryViewer = ({ initialImageKey }) => {
   )
 }
 
+const getLatLng = (imageInfo) => {
+  const lat = imageInfo.position?.lat !== undefined ? imageInfo.position.lat : imageInfo.lat;
+  const lon = imageInfo.position?.lon !== undefined ? imageInfo.position.lon : imageInfo.lon;
+  return (lat !== undefined && lon !== undefined) ? [lat, lon] : null;
+}
+
 const buildImageMarkers = (images, icon, markerColor, imageClicked, layerId, layerLabel) => {
   try {
     if (!images || images.length === 0) {
       return []
     }
 
     return _map(images, imageInfo => {
-      if (!imageInfo || (!imageInfo.position?.lat || !imageInfo.position?.lon) && (!imageInfo.lat || !imageInfo.lon)) {
+      const imageLatLon = getLatLng(imageInfo);
+      if (!imageInfo || !imageLatLon) {
         console.error(`Invalid position for image key: ${imageInfo?.key}`, imageInfo)
         return null
       }
 
+      if (imageLatLon.lat === undefined || imageLatLon.lon === undefined) {
+        console.error(`Invalid coordinates for image key: ${imageInfo?.key}`, imageInfo)
+        return null
+      }
+
```

Comment:
This seems like maybe a bug to me? `imageLatLon` is an array of two numbers `[lat, lon]` so the condition here will always be true, right?